backend/src/test/java/com/bakdata/conquery/util/support/StandaloneSupport.java

```diff
@@ -44,14 +52,27 @@ public void waitUntilWorkDone() {
 		testConquery.waitUntilWorkDone();
 	}
 
-	public void preprocessTmp() throws Exception {
-		DropwizardTestSupport<ConqueryConfig> prepro = new DropwizardTestSupport<>(
-			Conquery.class,
-			config,
-			app ->  new PreprocessorCommand(MoreExecutors.newDirectExecutorService())
-		);
-		prepro.before();
-		prepro.after();
+	public void preprocessTmp(File tmpDir, List<File> descriptions) throws Exception {
+		// Setup necessary mock
+		final JarLocation location = mock(JarLocation.class);
+		when(location.getVersion()).thenReturn(Optional.of("1.0.0"));
+
+		// Add commands you want to test
+		final Bootstrap<ConqueryConfig> bootstrap = TestBootstrappingConquery.createTestBootstrapConquery(tmpDir);
+
+		bootstrap.addCommand(new PreprocessorCommand(MoreExecutors.newDirectExecutorService()));
+
+		final Cli cli = new Cli(location, bootstrap, System.out, System.err);
+
+		final ImmutableList<String> params =
+				ImmutableList.<String>builder()
+							 .add("preprocess", "--in", tmpDir.toString(), "--out", tmpDir.toString())
+							 .add("--desc")
+							 .addAll(descriptions.stream().map(File::toString).collect(Collectors.toList()))
+							 .build();
+
+
+		cli.run(params.toArray(String[]::new));
```

Comment:
Ah krass, das ist jetzt deutlich aufwendiger.

Wäre es prinzipiell möglich die Preprocess methode direkt aufzurufen?

Reply:
Das createTestBootstrapConquery bringt deutlich Indirektion rein

Reply:
Factoryception

Reply:
Außer der FactoryFactory finde ich es so eigentlich schöner, weil wir jetzt in den Tests das machen, was wir auch in der Anwendung machen würden. Das Problem war, dass ich nur so einen weg gesehen habe die DW Namespace zu setzen.